scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -48,19 +51,42 @@
     (swap! map-atom dissoc key)
     (swap! map-atom assoc key value)))
 
-;; REVIEW: Review handle-status-update
+(defn container-status->failure-reason
+  [^V1PodStatus pod-status ^V1ContainerStatus status]
+  ; TODO map kubernetes failure reasons
+  (let [terminated (-> status .getState .getTerminated)]
+    (cond
+      (= "OutOfMemory" (.getReason pod-status)) :reason-container-limitation-memory
+      (= "Error" (.getReason terminated)) :reason-command-executor-failed
+      (= "OOMKilled" (.getReason terminated)) :reason-container-limitation-memory
+      :default :unknown)))
+
+(defn handle-status-update
+  [kcc mesos-status]
+  (scheduler/handle-status-update datomic/conn
+                                  @(:pool->fenzo-atom kcc)
+                                  mesos-status))
+
 (defn pod-has-just-completed
   "A pod has completed."
-  [{:keys [synthesized-state pod] :as existing-state-dictionary}]
-  (case (:state synthesized-state)
-    :pod/failed
-    (do
-      ; TODO: Extract failure reason, etc and store in datomic, via refactored handle-status-update.
-      {:expected-state :expected/completed})
-    :pod/succeeded
-    (do
-      ; TODO: Mark job as success in datomic, via refactored handle-status-update.
-      {:expected-state :expected/completed})))
+  [kcc {:keys [synthesized-state pod] :as existing-state-dictionary}]
+  (let [task-id (-> pod .getMetadata .getName)
+        ^V1ContainerStatus job-container-status (->> pod
```

Comment:
This smells like a helper function. Which also helps the indenting here.

Reply:
Done